sympy/matrices/matrices.py

```diff
@@ -266,10 +266,10 @@ def _eval_det_lu(self, iszerofunc=_iszero, simpfunc=None):
         # return det(P)*det(U)
         return det
 
-    def _eval_determinant(self):
+    def _eval_determinant(self, method=None):
         """Assumed to exist by matrix expressions; If we subclass
         MatrixDeterminant, we can fully evaluate determinants."""
-        return self.det()
+        return self.det(method=method)
```

Comment:
The problem here is that if method is None, it will replace "bareiss" in `det` and evaluation does not happen. We needed the default to be None earlier, because the function had to also deal with abstract Matrix Expressions, for which method doesn't make sense. But now that were are evaluating the determinant, None is no good: we should not pass it further.
The following should express this logic: 
```
if method is None:
     return self.det()
else:
     return self.det(method=method)
```